log/log.go

```diff
@@ -0,0 +1,114 @@
+// Package log provides logging functionality. To enable logging option should implement the
+// log.Logger interface and include it in context by calling log.WithLogger.
+// 3rd party loggers that implement log.Logger: github.com/uber-go/zap.SugaredLogger
+// and github.com/sirupsen/logrus.Logger.
+package log
```

Comment:
Yes, we should but since notation-go will depend on this package not other way around I had to create it here. Once we update notation-go to depend on this pkg we can remove it from notation-go

Reply:
> Yes, we should but since notation-go will depend on this package not other way around I had to create it here. Once we update notation-go to depend on this pkg we can remove it from notation-go

@priteshbandi Since `notation-go v1.0.0` is released, deleting packages such as  https://github.com/notaryproject/notation-go/blob/v1.0.0/plugin or https://github.com/notaryproject/notation-go/tree/v1.0.0/log is a breaking change. For example, both packages are used in Notation CLI, whose v1.0.0 is also released. /cc: @shizhMSFT
I think for the `notation-plugin-framework-go` repo, we could just include the boilerplate code required to generate plugin executables and the `example plugin`.

Reply:
That's a good point, probably the only time we would be able to remove duplication is 2.0 release.

Reply:
Yes but in this case if we want to simplify plugin generation we dont have any choice.

Reply:
I am still not convinced that the `log` package should be in the `notation-plugin-framework-go` package. The `log` package should either be in the `notation-go` package or be a standalone package.

What if we have yet another package named `notation-whatever-go` in the future and it requires the `log` package and is not related to `plugin`. Should `notation-whatever-go` create another `log` package and let `notation-plugin-framework-go` depend on `notation-whatever-go`? I think the answer is negative.

Reply:
Yes but there are only two options either we create new package call notation-utils-go which contains all utilities like logs or we duplicate log module in all the repo where we need them.

Reply:
As per discussion removed `Discard`, `WithLogger` and `GetLogger` methods and passed Logger as struct member in `cli`